content/posts/DevjitChoudhury_GSoC22_Phase2_Report.md

```diff
@@ -0,0 +1,125 @@
+
+#### VII. Export Verilog Code Dialog
+Create Verilog code for the circuit drawn and save it.
+![Export Verilog](/images/devjit_choudhury_GSoC'22/exportVerilog.gif)
+
+#### VIII. Save/Open Project Dialog
+Save the project offline and open it anytime later to continue working. It has some small bug with the tabs bar not able to render the new circuit list.
+![Open Project](/images/devjit_choudhury_GSoC'22/saveOffline.gif)
+
+### Canvas - Pinia Store
+Canvases in CircuitVerse Simulator -
+1. BackgroundArea Canvas - base canvas on which the grids are made
+2. SimulationArea Canvas - canvas on which the circuit elements are drawn
+3. MinimapArea Canvas - canvas showing the minimap of the entire area
+
+Initially, the canvas in the simulator is represented as an object.
+
+First, I had to figure out a way to implement the refactoring without breaking down the simulator since there are a lot of places where canvas objects are used.
+
+Since most of the usage of the canvases is outside the Vue components, I learned [#using a store outside of a component](https://pinia.vuejs.org/core-concepts/outside-component-usage.html). 
+
+
+## All the Pull Requests
+---
+- [Setting up the project and its dependencies](https://github.com/CircuitVerse/cv-frontend-vue/pull/2)
+- [Simulator codebase integration](https://github.com/CircuitVerse/cv-frontend-vue/pull/4)
+- [feat: circuit element panel component](https://github.com/CircuitVerse/cv-frontend-vue/pull/7)
+- [feat: properties panel component](https://github.com/CircuitVerse/cv-frontend-vue/pull/9)
+- [refactor: navbar component](https://github.com/CircuitVerse/cv-frontend-vue/pull/11)
+- [feat: tabsbar component](https://github.com/CircuitVerse/cv-frontend-vue/pull/14)
+- [feat: dialog box components](https://github.com/CircuitVerse/cv-frontend-vue/pull/20)
+- [feat: move canvas objects to pinia store](https://github.com/CircuitVerse/cv-frontend-vue/pull/25)
+
+
+> **[_GSoC22 Simulator UI Project -_](https://github.com/orgs/CircuitVerse/projects/12)
+> With this I have completed all the proposed task for VueJS Simulator Project. Some essential tasks that were added during the coding period will be done next.**
+
+### Future Works -
+---
+This project provides a base for the VueJS Simulator. 
+Though I checked for bugs and solved them while working on the project but still there might be many that I might have missed.
+- The most important task now is to integrate this Vue Project into the Main Repository and have an option to switch between the Simulators.
+- After the integration - check if the saving online and loading from the server is working.
+- Embed Feature - Planning a way to integrate the embed feature into the Vue Simulator.
+- Progressive work on Internationalization of Simulator and Refactoring of Styles.
+- Few components - Verilog Module, Timing-Diagram, Quick-Button panel, and Testbench are yet to be converted to Vue.
```

Comment:
Could you expand this part in more detail, also include missed tasks like API integration, removing jquery (jquery ui is removed, we still have jquery) etc

Reply:
okay

Reply:
created a PR for it, can you please check -
https://github.com/CircuitVerse/Blog/pull/151